src/utils/common-utils.test.ts

```diff
@@ -0,0 +1,173 @@
+import { flattenObsList, hasRendering, clearSubmission, gracefullySetSubmission, hasSubmission } from './common-utils';
+import { isEmpty } from '../validators/form-validator';
+import { type FormField, type OpenmrsObs } from '../types';
+
+jest.mock('@openmrs/esm-framework', () => ({
+  formatDate: jest.fn(),
+  restBaseUrl: 'http://openmrs.com/rest',
+}));
+
+jest.mock('../validators/form-validator', () => ({
+  isEmpty: jest.fn(),
+}));
+
+describe('utils functions', () => {
+  describe('flattenObsList', () => {
+    it('should flatten a nested obs list', () => {
+      const obsList: OpenmrsObs[] = [
```

Comment:
We should consider moving this to the mocks directory just to have a cleaner code base.

Reply:
thanks @arodidev 
Done